aws/resource_aws_budgets_budget.go

```diff
@@ -0,0 +1,431 @@
+package aws
+
+import (
+	"fmt"
+	"log"
+	"strings"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/budgets"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func resourceAwsBudgetsBudget() *schema.Resource {
+	return &schema.Resource{
+		Schema: resourceAwsBudgetsBudgetSchema(),
```

Comment:
Is there a reason for the indirection here? (e.g. reusing the schema for some other resource) Generally we prefer to inline it unless its reused multiple times.

Reply:
yeah, I'm beginning to realize that with the mono-package approach, minimizing the number of functions really reduces the overall package noise and potential for namespace conflict.

Do you feel there are benefits with one approach over the other (package per resource vs. mono-package), ie. are there reasons a single package approach was chosen originally?

Reply:
I can't speak for the original implementation (way before my time with the codebase and HashiCorp), but generally we lean towards mono-package nowadays since it tends to reduce friction for new contributors and less debates about code boundaries.